src/query-builder/QueryPartialEntity.ts

```diff
@@ -11,7 +9,7 @@ export type QueryPartialEntity<T> = {
  * Make all properties in T optional. Deep version.
  */
 export type QueryDeepPartialEntity<T> = _QueryDeepPartialEntity<
-    ObjectLiteral extends T ? unknown : T
+    T extends "toString" ? T : unknown
```

Comment:
For me it seems like

```ts
/**
 * Make all properties in T optional
 */
export type QueryPartialEntity<T> = {
    [P in keyof T]?: T[P] | (() => string)
}
```

could be an issue.
I would have expect something like

```ts
/**
 * Make all properties in T optional
 */
export type QueryPartialEntity<T> = {
    [P in keyof T]?: QueryPartialEntity<T[K]>
}
```

for Deep partial objects, or the long version

```ts
interface DeepPartialArray<T> extends Array<DeepPartial<T>> {}

export type QueryPartialEntityObject<T> = {
    [P in keyof T]?: QueryPartialEntity<T[K]>
}
export type QueryPartialEntity<T> = T extends Function
  ? T
  : T extends Array<infer InferredArrayMember>
  ? DeepPartialArray<InferredArrayMember>
  : T extends ObjectLiteral
  ? QueryPartialEntityObject<T>
  : T | undefined;
```

props to @mattpocockuk

https://www.youtube.com/watch?v=AhzjPAtzGTs